contracts/ics721/src/contract.rs

```diff
@@ -149,22 +153,24 @@ fn query_list(deps: Deps) -> StdResult<ListChannelsResponse> {
 
 pub fn query_channel(deps: Deps, id: String) -> StdResult<ChannelResponse> {
     let info = CHANNEL_INFO.load(deps.storage, &id)?;
-
     let _class_ids: StdResult<Vec<_>> = CHANNEL_STATE
         .sub_prefix(&id)
         .range(deps.storage, None, None, Order::Ascending)
         .map(|r| {
             let (class_id_token_id, _) = r?;
-            // TODO: extract class_id out of Vec<u8>
-            // https://github.com/public-awesome/contracts/issues/60
-            Ok(class_id_token_id)
+            Ok(class_id_token_id.0)
         })
-        // TODO: filter duplicates
         .collect();
 
+    let mut class_ids_resp = _class_ids.unwrap();
```

Comment:
I would unwrap the response with `?` after `collect()`, and remove this `unwrap()` so the error can be handled.

Reply:
Now doing a match: 

```
    let class_ids_resp = _class_ids;
    match class_ids_resp {
        Ok(mut class_id_vec) => Ok(ChannelResponse {
            info,
            class_ids: {
                class_id_vec.sort();
                class_id_vec.dedup();
                class_id_vec
            },
        }),
        Err(msg) => Err(StdError::GenericErr { msg: msg.to_string() })
 ```